src/main/java/com/metamx/emitter/service/ServiceMetricEvent.java

```diff
@@ -250,6 +256,12 @@ public Builder setUser10(String x)
       return this;
     }
 
+    public Builder setDimension(String dim, String value)
+    {
+      userDims.put(dim, value);
```

Comment:
There's a lot of code in https://github.com/druid-io/druid/pull/995 for handling nulls in dim/value. Would those ever be expected to propagate to this layer?